### PR TITLE
chore: add vesting events

### PR DESCRIPTION
**Pull request by crystalbit**

### CODE REVIEW

File: src/core/Vesting.sol [hunk 1]
```diff
@@ -88,19 +91,26 @@ library Vesting {
         }
 
         for (uint256 i = 0; i < vestingPeriodCount; i++) {
+            uint256 amount = rewardPerPeriod * (i + 1);
```
[the remark this answers is not in the document]
Comment (ValeryAntopol) [reply src/core/Vesting.sol:94]: That is because we are counting cumulative sums. `cumulateiveAmountIncrease` may be better name

Comment (folex) [inline src/core/Vesting.sol:94]: Let's call it `cumulativeAmount`, then.
```suggestion
            uint256 cumulativeAmount = rewardPerPeriod * (i + 1);
```

File: src/core/Vesting.sol [hunk 2]
```diff
@@ -30,6 +30,9 @@ library Vesting {
         uint256 cumulativeAmount;
     }
 
+    event VestingScheduleItem(uint256 indexed id, uint256 indexed epoch, uint256 amount, uint256 cumulativeAmount);
+    event VestingUpdateItem(uint256 indexed id, uint256 amount, uint256 cumulativeAmount);
+
```
Comment (ValeryAntopol) [inline src/core/Vesting.sol:35]: I'm not sure if the indexer can use the `amount` field meaningfully. But anyway, this should be fine

File: src/core/Vesting.sol [hunk 3]
```diff
@@ -88,19 +91,26 @@ library Vesting {
         }
 
         for (uint256 i = 0; i < vestingPeriodCount; i++) {
+            uint256 amount = rewardPerPeriod * (i + 1);
             // add reward to cumulativeAmount
```
Comment (folex) [inline src/core/Vesting.sol:95]: And I guess this comment should be rewritten, right

**Review by ValeryAntopol — APPROVED** [review]
utACK

**Review by folex — APPROVED** [review]

File: src/core/Vesting.sol [hunk 4]
```diff
@@ -88,19 +91,27 @@ library Vesting {
         }
 
         for (uint256 i = 0; i < vestingPeriodCount; i++) {
-            // add reward to cumulativeAmount
+            uint256 cumulativeAmountIncrease = rewardPerPeriod * (i + 1);
```
Comment (folex) [inline src/core/Vesting.sol:94]: I'd name it without `increase`, as it misleads me into thinking that it is a per-period-delta. But it's the whole `cumulativeAmount` for the whole `rewardPeriod`.

**Pull request merged**